spec/controllers/projects_controller_spec.rb

```diff
@@ -1,13 +1,85 @@
 require 'spec_helper'
 
+RSpec.shared_examples 'denies access' do
```

Comment:
I agree with you. The reason I have added the shared examples in this test is because I didn't want to duplicate the same code for almost all the examples.